File: fizzbuzz.py
def fizzbuzz(n):
    ret = ""
    if n % 3 == 0:
        ret = "Fizz"
    if n % 5 == 0:
        ret += "Buzz"
    if ret == "":
        ret = n
    return ret


def print_fizzbuzz_1_100(display_function):
    for i in range(1, 101):
        display_function(fizzbuzz(i))


print_fizzbuzz_1_100(print)

File: googletest_projects/exercise4/Sources/fizzbuzz.c
#include <string.h>
#include <stdio.h>
#include "fizzbuzz.h"

char rsp_str[10];

char* fizzbuzz( int number ) {
    /* Not multiple of 5 or 3 */
    if ((number % 5) && (number % 3))
    {
        sprintf(rsp_str, "%d", number );
        return rsp_str;
    }
    if (((number % 3) == 0) && ((number % 5) == 0)) {
        sprintf(rsp_str, "fizzbuzz" );
        return rsp_str;
    }
    if ((number % 3) == 0) {
        sprintf(rsp_str, "fizz" );
        return rsp_str;
    }
    if ((number % 5) == 0) {
        sprintf(rsp_str, "buzz" );
        return rsp_str;
    }
}